File: documentation/user/en/query/examples/complexGrammar.cs
Query(
    Collection("Product"),
    FilterBy(
       And(
          EntityPrimaryKeyInSet(1, 2, 3),
          AttributeEquals("visibility", "VISIBLE")
       )
    ),
    OrderBy(
        AttributeNatural("code", Asc),
        AttributeNatural("priority", Desc)
    ),
    Require(
        EntityFetch(
			AttributeContentAll(), PriceContentAll()
		),
        FacetSummary()
    )
)

File: documentation/user/en/query/examples/complexGrammar.java
final EvitaResponse<SealedEntity> entities = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				head(
					collection("Product"),
					label("query-name", "my-query")
				),
				filterBy(
					and(
						entityPrimaryKeyInSet(1, 2, 3),
						attributeEquals("status", "ACTIVE")
					)
				),
				orderBy(
					attributeNatural("code", ASC),
					attributeNatural("catalogNumber", DESC)
				),
				require(
					entityFetch(
						attributeContentAll(),
						priceContentAll()
					),
					facetSummary(COUNTS)
				)
			)
		);
	}
);